python/pyiceberg/avro/decoder.py

```diff
@@ -191,3 +196,136 @@ def skip_bytes(self) -> None:
 
     def skip_utf8(self) -> None:
         self.skip_bytes()
+
+
+class StreamingBinaryDecoder(BinaryDecoder):
```

Comment:
Should we remove this one and combine everything back into the decoder? I like to keep things simple.

Reply:
It depends.  It is useful to have a decoder that can work on a streaming basis and it's good to test subsequent implementations with.  It may also be useful for a "pure python" approach to keep it around.

Reply:
> It is useful to have a decoder that can work on a streaming basis and it's good to test subsequent implementations with.

Should we move it to tests then? It is not used here. Preferable in such a way that we merge the InMemoryDecoder into the Decoder itself, and then we can subclass that one with the streaming decoder.

> It may also be useful for a "pure python" approach to keep it around.

For Python we use the `InMemoryDecoder` as well, right?